run-custom-mao.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash -xe
+
+# This script needs an machine-api-operator image as an input parameter
+# For example: ./run-custom-mao.sh quay.io/sdasu/machine-api-operator:sdasu-fix
+export SCRIPTS_DIR=$(pwd)
+export CUSTOM_MAO_BUILD=$1
+
+kubectl get secrets --namespace openshift-machine-api | grep mariadb
+
+# Stop CVO, MAO, machine-api-controller and Metal3 deployments
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment --replicas=0 cluster-version-operator --namespace openshift-cluster-version
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment -n openshift-machine-api --replicas=0 machine-api-operator
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment --replicas=0 machine-api-controllers --namespace openshift-machine-api
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment --replicas=0 metal3 --namespace openshift-machine-api
+
+REPLACEMENT_TEXT="image: ${CUSTOM_MAO_BUILD}"
+echo $REPLACEMENT_TEXT
+
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig get deployments -n openshift-machine-api machine-api-operator -o yaml > mao.yaml
+sed -i -e "s!image: .*!${REPLACEMENT_TEXT}!g" mao.yaml
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig apply -f mao.yaml
+
+# Give the machine-api-operator rbac access to create and read secrets
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig apply -f machine-api-operator-roles.yaml
```

Comment:
Instead of having this in dev-scripts, how about making `run-custom-mao` apply the latest manifests from the MAO based on the branch it's using?  It seems like this could make CI work, but it would break in a normal deployment if it's missing some required rbac?

Reply:
Agreed. We could make this script apply all manifests from a given branch to make it more generic.